atc/exec/set_pipeline_step.go

```diff
@@ -130,7 +130,40 @@ func (step *SetPipelineStep) run(ctx context.Context, state RunState) error {
 		return nil
 	}
 
-	team := step.teamFactory.GetByID(step.metadata.TeamID)
+	var team db.Team
+	if step.plan.Team == "" {
+		team = step.teamFactory.GetByID(step.metadata.TeamID)
+	} else {
+		fmt.Fprintln(stderr, "\x1b[1;33mWARNING: specifying the team in a set_pipeline step is experimental and subject to change!\x1b[0m")
+		fmt.Fprintln(stderr, "")
+		fmt.Fprintln(stderr, "\x1b[33mcontribute to discussion #5731 with feedback: https://github.com/concourse/concourse/discussions/5731\x1b[0m")
+		fmt.Fprintln(stderr, "")
+
+		currentTeam, found, err := step.teamFactory.FindTeam(step.metadata.TeamName)
```

Comment:
To avoid an extra `FindTeam` query here, can we change the original team assignment (using `GetByID`) to `currentTeam` and then do `team = currentTeam` if `step.plan.Team` is not set?

i.e.:

```go
currentTeam := step.teamFactory.GetByID(step.metadata.TeamID)
if step.plan.Team == "" {
  team = currentTeam
} else {
  // ...
}
```

Reply:
AFAICT this represents a team, but doesn't make the team functional for use in the following checks

If we use `GetByID` then the `currentTeam.Admin()` call will return false, because it is the zero value of the struct

Reply:
Blargh, so it's partially instantiated? That's annoying. OK with this as-is then.